### DISCUSSION TITLE
Is there any full documentation of the API?

### DISCUSSION THREAD

**Discussion by Stokestack** [supabase/supabase#28107] [Questions]
I'm tying to adopt Supabase in my Deno-based project, but there's a lot of guesswork involved because of undefined terms in the documentation [here](https://supabase.com/docs/reference/javascript/initializing).

For example:
<img width="425" alt="Screenshot 2024-07-20 at 5 10 21 PM" src="https://github.com/user-attachments/assets/140cf5d9-a2bb-4fc8-bf8b-6466758fe714">

I don't know what to make of "a storage provider." Is there really no further documentation on that, or of the many other similar terms used throughout this doc? Also I don't see any documentation on query syntax here. I am perplexed as to how this can work, for example:
<img width="462" alt="Screenshot 2024-07-20 at 5 13 20 PM" src="https://github.com/user-attachments/assets/7e94e6eb-639a-4fe2-ac66-c93578c25d04">

Any insight appreciated!

**Comment by GaryAustin1** [comment]
If you don't find it in docs with the search, then there are the repositories for all the client code.

The storage provider is local storage on a browser.  Normally you would not use the option to intercept that on Deno as you are not saving session info.

As far as the JS format for queries, most of the doc guides and database section have examples in JS.  There is also the REST API Supabase uses documentation at https://docs.postgrest.org/en/v12/references/api.html .  It has lots of details, but is the underlying http url format so some conversion needed to supabase-js, which simplifies it.

There is also the API tab in the dashboard which gives your API call examples for your tables, etc.

**Reply by Stokestack** [reply]
Thanks! Correct; obviously there is no browser in my case. But if there were, I still would have nothing to go on as to how to specify a storage provider.

The JS examples are wanting, which I've tried to illustrate above with those examples. I'm not sure you can guess the JavaScript syntax from the REST examples. I'm not using the dashboard at all, since I'm doing all local CLI-based development. I guess I'll have to resort to using it if that's the only way to get insight to the query format. At the moment, I'm dead in the water because of the "window is not defined" bug: https://github.com/supabase/supabase-js/issues/1249
